steve/runner.go

```diff
@@ -178,7 +208,10 @@ func (r *runner) Stop(ctx context.Context, id ID) error {
 	j := obj.(*jobIO)
 
 	// Ignore if already stopped
-	if atomic.LoadInt64(&j.running) == 0 {
+	j.RLock()
+	running := j.status.Running
+	j.RUnlock()
```

Comment:
> Mutexes are slow, due to the setup and teardown, and due to the fact that they block other goroutines for the duration of the lock.
>
> Atomic operations are fast because they use an atomic CPU instruction, rather than relying on external locks to.
>
> Therefore, whenever it is feasible, atomic operations should be preferred.

https://stackoverflow.com/a/47446114